### ISSUE TITLE
`assert_display_snapshot` vs. `assert_snapshot`

### ISSUE THREAD

**Issue by max-sixty** [mitsuhiko/insta#379]
Not the biggest deal — but would it simplify the API if we combined these? 

IIUC if we pass anything that's not a `&str` to `assert_snapshot`, it won't compile. `assert_display_snapshot`'s only different behavior is to coerce anything that's not a `&str` (which is `ToString`) to a string.

So would it make sense to:
- Rename `assert_display_snapshot` to `assert_snapshot`
- Deprecate `assert_display_snapshot`

...and then it's like the type of the value is anything that implements `ToString`?

**Comment by mitsuhiko** [comment]
That sounds reasonable.

**Comment by cosmincatalin** [comment]
Beginner with insta here, but `assert_snapshot` does not seem to have feature parity with `assert_display_snapshot`.

In my case, I want to snapshot a `GraphQL` query that is multiline for which no other `assert_*` creates a proper snapshot, other than `assert_display_snapshot`. As it looks to me `assert_snapshot` is fine, but does not create a snapshot file, which is really the main point.

For reference, here is how a `GraphQL` query looks like:

```graphql
---
source: tests/snapshots.rs
assertion_line: 43
expression: table.gql_query()
---
query HeroNameAndFriends($episode: Episode) {
  hero(episode: $episode) {
    name
    friends {
      name
    }
  }
}
```

**Comment by max-sixty** [comment]
> multiline for which no other `assert_*` creates a proper snapshot, other than `assert_display_snapshot`. As it looks to me `assert_snapshot` is fine, but does not create a snapshot file, which is really the main point.

Could you show the rust? 

File vs. inline snapshots are orthogonal to the snapshot macro, check out the docs for more info

**Comment by cosmincatalin** [comment]
The output above is obtained by running something similar to this:

```rust
    with_settings!({snapshot_suffix => format!("{}-{}", env, query_name)}, {
        assert_display_snapshot!("schema_graphql", gql_query);
    });
```

I have studied the docs, but maybe I have missed some information; can I replace the above with just `assert_display` and get the same result (actual snapshot file)?

**Comment by max-sixty** [comment]
> can I replace the above with just `assert_display` and get the same result (actual snapshot file)?

Yes, with `assert_snapshot`; what different result do you get when you do that?

**Comment by cosmincatalin** [comment]
Seems to be working fine if I just make the replacement.
I must have done something wrong before.
I apologize for the false alarm and thank you for your support.
